python/tank/authentication/user_impl.py

```diff
@@ -206,7 +206,19 @@ def __init__(self, host, login, session_token, http_proxy, password=None, cookie
         self._login = login
         self._session_token = session_token
         self._cookies = cookies
-        self._saml_expiration = saml_expiration
+
+        # saml expiration is meaningful only with SSO enabled.
+        # The presence of cookies implies potential SSO usage.
+        if self._cookies is None:
+            self._saml_expiration = None
+        elif saml_expiration is None:
+            # For lack of accurate information, we will indicates that the
+            # credentials are expired  by setting the expiration to a moment
+            # in the past.
+            # @TODO: obtain the expiration from the cookies themselves, if possible.
```

Comment:
I did not do it now because I wanted to make only the minimal changes needed to get the claims renewal going. 

Setting the initial expiration from the cookies is only a marginal optimisation. At best, we realize that the claims are still valid and we do not need to immediately renew them, at worst we need to make one renewal loop. In the scheme of things, not a biggie.

The real win in getting the expiration from the cookies is to ensure that we are not provided with an incorrect saml_expiration value (e.g. saying that the claims would expire in 5 months).

Once we expose the shotgun cookies with a proper API, I would remove most of the places where an explicit saml_expiration value is passed as an argument to a function/method.

Since we have already talked about a pre-merge-in-master refactoring for the unsightly 5-object tuple used in a few places, I was thinking of exposing the cookies then (at the latest) or before.